packages/api/src/services/tags.service.ts

```diff
@@ -47,4 +48,17 @@ export class TagsService {
 
         return this.tagRepository.findOneBy({ userId, tagName: newName });
     }
+
+    async delete(deleteTagRequest: DeleteTagRequest): Promise<TransactionTag> {
+        const userId = deleteTagRequest.id;
+        const tagName = deleteTagRequest.name;
+        const tag = await this.tagRepository.findOneBy({ userId, tagName });
+        if (!tag) {
+            throw new BadRequestException(ERROR_TAG_NOT_FOUND);
+        }
+
+        this.tagRepository.delete({ userId, tagName });
+
+        return tag;
```

Comment:
I think you want to do it like this, the other way is fine too just make sure you await the delete of the tag so the UI doesn't have any weird side effects

```suggestion
        return this.tagRepository.remove(tag);
```